pagination/pagetoken.go

```diff
@@ -32,8 +32,12 @@ func ParsePageToken(request Request) (_ PageToken, err error) {
 	}
 	requestChecksum ^= pageTokenChecksumMask // apply checksum mask for PageToken
 	if request.GetPageToken() == "" {
+		offset := int64(request.GetPageSize())
+		if s, ok := request.(skipRequest); ok {
+			offset += int64(s.GetSkip())
+		}
 		return PageToken{
-			Offset:          0,
```

Comment:
I must be missing something, I feel like my change to this should be what we want? The page token should point to `pageSize` + possible `skip`. Not just 0?